reference/src/representation/function-pointers.md

```diff
@@ -0,0 +1,116 @@
+# Representation of Function Pointers
+
+### Terminology
+
+In Rust, a function pointer type, is either `fn(Args...) -> Ret`,
+`extern "ABI" fn(Args...) -> Ret`, `unsafe fn(Args...) -> Ret`, and
+`unsafe extern "ABI" fn(Args...) -> Ret`.
+A function pointer is the address of a function,
+and has function pointer type.
+The pointer is implicit in the `fn` type,
+and `fn` types are implicitly `'static`.
```

Comment:
I'm not sure what it means to be `'static`. If "`T` is 'static" means `T: 'static`, then this is not *strictly* true. For example,

```rust
fn(&'a u32): 'static
```

does not hold. This is fallout from the "syntactic definition of outlives". Such a type is not entirely trivial to make though.

I think the point is that fn pointer types do not carry their own lifetime, so they are always assumed to point to a statically defined function with no state of its own, right? Maybe we can word that more directly.